cms/utils/placeholder.py

```diff
@@ -39,32 +40,44 @@ def get_placeholder_conf(setting, placeholder, template=None, default=None):
     Returns the placeholder configuration for a given setting. The key would for
     example be 'plugins' or 'name'.
 
-    If a template is given, it will try
-    CMS_PLACEHOLDER_CONF['template placeholder'] and
-    CMS_PLACEHOLDER_CONF['placeholder'], if no template is given only the latter
-    is checked.
+    Resulting value will be the last from:
+
+    CMS_PLACEHOLDER_CONF[None] (global)
+    CMS_PLACEHOLDER_CONF['template'] (if template is given)
+    CMS_PLACEHOLDER_CONF['placeholder']
+    CMS_PLACEHOLDER_CONF['template placeholder'] (if template is given)
     """
+
     if placeholder:
         keys = []
+        placeholder_conf = get_cms_setting('PLACEHOLDER_CONF')
+        # 1st level
+        if template:
+            keys.append(force_text('%s %s' % (template, placeholder)))
+        # 2nd level
+        keys.append(force_text(placeholder))
+        # 3rd level
         if template:
-            keys.append("%s %s" % (template, placeholder))
-        keys.append(placeholder)
+            keys.append(force_text(template))
+        # 4th level
+        keys.append(None)
         for key in keys:
-            conf = get_cms_setting('PLACEHOLDER_CONF').get(key)
-            if not conf:
-                continue
-            value = conf.get(setting)
-            if value is not None:
-                return value
-            inherit = conf.get('inherit')
-            if inherit:
-                if ' ' in inherit:
-                    inherit = inherit.split(' ')
-                else:
-                    inherit = (None, inherit,)
-                value = get_placeholder_conf(setting, inherit[1], inherit[0], default)
-                if value is not None:
-                    return value
+            for conf_key, conf in iteritems(placeholder_conf):
```

Comment:
We should write new code for Py3 and shim it for Py2. By using six.iteritems, we're actually doing the opposite. I suggest to just use plain items(). If this was in a tight loop of many items, there would be grounds for doing something else, but this will not be *that* inefficient in Py2 to make it worthwhile.